File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Teoria/T4/ejemplostema4java/src/ConstructorCopiaSuperficial/PruebaConstructorCopiaSuperficial.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ConstructorCopiaSuperficial;

/**
 *
 * @author ana
 */
// prueba constructor de copia superficial
public class PruebaConstructorCopiaSuperficial {
   public static void main(String argsString[]) throws CloneNotSupportedException {

    Direccion d1= new Direccion("Calle del Agua", 58, "18888", "Murcia");

    Persona p1 = new Persona("Juan", 30, d1); 
    Persona p2 = new Persona(p1);
    System.out.println(p1.nombre);
    p2.nombre = "Luis";     
    System.out.println(p1.nombre); 

    System.out.println("La ciudad de " + p2.nombre + " es " + p2.dir.ciudad);
    p2.dir.ciudad = "Huelva";
    // el cambio de dirección afecta también a p1, porque en la copia 
    // superficial se crea otra referencia a la dirección (al mismo objeto)
    System.out.println("La ciudad de " + p1.nombre + " es " + p1.dir.ciudad);
    System.out.println("La ciudad de " + p2.nombre + " es " + p2.dir.ciudad);
     
    }  
}

File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Teoria/T4/ejemplostema4ruby/lib/ConstructorCopiaSuperficial/PruebaConstructorCopiaSuperficial.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
  require_relative 'direccion.rb'
  require_relative 'persona.rb'
  d1= Direccion.new("Calle del Agua", 58, "18888", "Murcia")

  p1 = Persona.new("Juan", 30, d1)
  p2 = Persona.newCopia(p1)
  puts p1.nombre
  p2.nombre = "Luis"   
  puts p1.nombre

  puts "La ciudad de #{p2.nombre} es #{ p2.direccion.ciudad}"
  p2.direccion.ciudad = "Huelva"
  puts "La ciudad de #{p1.nombre} es #{ p1.direccion.ciudad}"
  puts "La ciudad de #{p2.nombre} es #{ p2.direccion.ciudad}" 